flexget/plugins/services/myepisodes.py

```diff
@@ -88,99 +86,154 @@ class MyEpisodes(object):
         'additionalProperties': False
     }
 
+
     @plugin.priority(-255)
     def on_task_output(self, task, config):
-        """Mark all accepted episodes as acquired on MyEpisodes"""
+        """
+        Mark all accepted episodes as acquired on MyEpisodes
+        """
+
         if not task.accepted:
             # Nothing accepted, don't do anything
             return
 
-        username = config['username']
-        password = config['password']
-
-        cookiejar = http.cookiejar.CookieJar()
-        opener = request.build_opener(request.HTTPCookieProcessor(cookiejar))
-        baseurl = request.Request('http://www.myepisodes.com/login.php?')
-        loginparams = parse.urlencode({'username': username,
-                                       'password': password,
-                                       'action': 'Login'})
         try:
-            logincon = opener.open(baseurl, loginparams)
-            loginsrc = logincon.read()
-        except URLError as e:
-            log.error('Error logging in to myepisodes: %s' % e)
-            return
+            self.plugin_config = config
+            self.db_session = task.session
+            self.test_mode = task.options.test
 
-        if str(username) not in loginsrc:
-            raise plugin.PluginWarning(('Login to myepisodes.com failed, please check '
-                                        'your account data or see if the site is down.'), log)
+            # attempt authentication
+            self.http_session = self._login(config)
+
+        except plugin.PluginWarning as w:
+            log.warning(str(w))
+            return;
```

Comment:
Nope. It effectively returns exits the method and returns None. If you feel it causes a readability issue, I will change it.